src/RefractAPI.cc

```diff
@@ -7,17 +7,450 @@
+    {
+        refract::MemberElement* element = new refract::MemberElement;
+
+        element->meta["classes"] = CreateArrayElement("user");
+        element->meta["classes"].renderType(refract::IElement::rCompact);
+        element->set(refract::IElement::Create(metadata.first), refract::IElement::Create(metadata.second));
+        element->renderType(refract::IElement::rFull);
+
+        return element;
+    }
+
+    refract::IElement* MetadataCollectionToRefract(const snowcrash::MetadataCollection metadata)
+    {
+        refract::ArrayElement* element = new refract::ArrayElement;
+        RefractElements content;
+
+        std::transform(metadata.begin(), metadata.end(), std::back_inserter(content), MetadataToRefract);
+        element->set(content);
+        element->renderType(refract::IElement::rCompact);
+
+        return element;
+    }
+
+    refract::IElement* CopyToRefract(const std::string& copy)
+    {
+        if (copy.empty()) {
+            return NULL;
+        }
+
+        refract::IElement* element = refract::IElement::Create(copy);
+        element->element("copy");
+
+        return element;
+    }
+
+    template<typename T>
+    refract::IElement* ParameterValuesToRefract(const snowcrash::Parameter& parameter)
+    {
+        refract::ArrayElement* element = new refract::ArrayElement;
+        RefractElements content;
+
+        element->element("enum");
+
+        if (!parameter.exampleValue.empty()) {
+            element->attributes["samples"] = CreateArrayElement(LiteralTo<T>(parameter.exampleValue));
+        }
+
+        for (snowcrash::Values::const_iterator it = parameter.values.begin();
+             it != parameter.values.end();
+             ++it) {
+
+            content.push_back(refract::IElement::Create(LiteralTo<T>(*it)));
+        }
+
+        element->set(content);
+
+        return element;
+    }
+
+    refract::IElement* ParameterToRefract(const snowcrash::Parameter& parameter)
+    {
+        refract::MemberElement* element = new refract::MemberElement;
+        refract::IElement *value, *defaultValue;
+
+        // Parameter type
+        if (parameter.type == "boolean") {
+            value = refract::IElement::Create(LiteralTo<bool>(parameter.exampleValue));
+            defaultValue = refract::IElement::Create(LiteralTo<bool>(parameter.defaultValue));
+        }
+        else if (parameter.type == "number") {
+            value = refract::IElement::Create(LiteralTo<double>(parameter.exampleValue));
+            defaultValue = refract::IElement::Create(LiteralTo<double>(parameter.defaultValue));
+        }
+        else {
+            value = refract::IElement::Create(parameter.exampleValue);
+            defaultValue = refract::IElement::Create(parameter.defaultValue);
+        }
+
+        // Values
+        if (!parameter.values.empty()) {
+            if (parameter.type == "boolean") {
+                value = ParameterValuesToRefract<bool>(parameter);
```

Comment:
probably memory leak - there will be content of `value` lost if it was already set on line  108. dtto for all other assigns below.